### DISCUSSION TITLE
PopupBox in modal mode?

### DISCUSSION THREAD

**Discussion by DrCQ** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2823] [Q&A] [answered]
I'm using PopupBox to open a dialog (UserControl) written with MVVM patterns. The StaysOpen property is set to True, but any click outside the PopupBox content is closing my dialog, The same happens when I try to examine the dialog properties using Blend. Click on the PopupBox in the Blend application is also closing the PopupBox content. It is possible to set the PopupBox in the modal mode?

**Comment by Keboo** [comment, marked as answer]
This is probably a bit of a poor name for `StaysOpen` on `PopupBox`. The `PopupBox` control is designed to be a simple wrapper around a toggle button and a popup control. Unfortunately, the `StaysOpen` property means that the popup should stay open even after the user interacts with it, it is not the same as the `Popup.StaysOpen` property. Effectively this control is meant for short quick interactions, not longer, dialog-style implementations.  

Unfortunately, there are only a few options I can think of:
- Switch to using the `DialogHost` control this one also uses a `Popup` but it is quite a bit different so might not mean your UX.
- Roll your own implementation with a `Popup` and `ToggleButton` controls. Since `PopupBox` is just a thin wrapper around these controls it would let you have more control over the behavior.
